### DISCUSSION TITLE
Beim Speichern eines Formulares/Datensatzes bekomme ich "$id has to be an integer greater than 0, but "-1" given"

### DISCUSSION THREAD

**Discussion by tyrant88** [FriendsOfREDAXO/.github#34] [Q&A] [answered]
InvalidArgumentException | $id has to be an integer greater than 0, but "-1" given | redaxo/src/addons/yform/plugins/manager/lib/yform/manager/dataset.php.

Was kann das sein?

**Comment by rkemmere** [comment]
Ist fast so wie "allgemeiner Datenbank Fehler".

Hatte es heute das Problem, dass in einem Projekt der Submit Button an einer anderen Stelle das no_db Flag hatte, als in anderen Installationen.

**Comment by tyrant88** [comment, marked as answer]
Ah du meinst es müsste so:
`$yform->setValueField('submit', ['submit','senden','','1']);`
und die '1' ist das no_db ?

für die "Pipetisten":
`submit|submit|senden||1|`

**Reply by rkemmere** [reply]
Bei dem Problemprojekt war es: 

`$yform->setValueField('submit', ['submit', 'senden', '', '', 'no_db', '']);`

**Reply by rkemmere** [reply]
Also genau eine Parameter Stelle weiter.

**Reply by rkemmere** [reply]
in der Doku steht:

`$yform->setValueField('submit', array('submit','Anfrage senden','','','','btn-secondary'));`

ohne das no_db

**Reply by rkemmere** [reply]
also try and error.

**Comment by tyrant88** [comment]
dann sollte das "no_db" besser in die Doku aufgenommen werden.

**Comment by rkemmere** [comment]
Hatte heute den gleichen Fehler, wobei einfach nur die beiden E-Mail Templates nicht vorhanden waren. Also die Templates waren zwar im Formular drin, aber in YFORM nicht angelegt.
